pkg/registry/flowcontrol/rest/storage_flowcontrol.go

```diff
@@ -0,0 +1,148 @@
+	serverstorage "k8s.io/apiserver/pkg/server/storage"
+	flowcontrolclient "k8s.io/client-go/kubernetes/typed/flowcontrol/v1alpha1"
+	"k8s.io/klog"
+	"k8s.io/kubernetes/pkg/api/legacyscheme"
+	"k8s.io/kubernetes/pkg/apis/flowcontrol"
+	"k8s.io/kubernetes/pkg/apis/flowcontrol/bootstrap"
+	flowschemastore "k8s.io/kubernetes/pkg/registry/flowcontrol/flowschema/storage"
+	prioritylevelconfigurationstore "k8s.io/kubernetes/pkg/registry/flowcontrol/prioritylevelconfiguration/storage"
+)
+
+const PostStartHookName = "apiserver/bootstrap-system-flowcontrol-configuration"
+
+type RESTStorageProvider struct{}
+
+var _ genericapiserver.PostStartHookProvider = RESTStorageProvider{}
+
+func (p RESTStorageProvider) NewRESTStorage(apiResourceConfigSource serverstorage.APIResourceConfigSource, restOptionsGetter generic.RESTOptionsGetter) (genericapiserver.APIGroupInfo, bool) {
+	apiGroupInfo := genericapiserver.NewDefaultAPIGroupInfo(flowcontrol.GroupName, legacyscheme.Scheme, legacyscheme.ParameterCodec, legacyscheme.Codecs)
+
+	if apiResourceConfigSource.VersionEnabled(flowcontrolv1alpha1.SchemeGroupVersion) {
+		apiGroupInfo.VersionedResourcesStorageMap[flowcontrolv1alpha1.SchemeGroupVersion.Version] = p.v1alpha1Storage(apiResourceConfigSource, restOptionsGetter)
+	}
+	return apiGroupInfo, true
+}
+
+func (p RESTStorageProvider) v1alpha1Storage(apiResourceConfigSource serverstorage.APIResourceConfigSource, restOptionsGetter generic.RESTOptionsGetter) map[string]rest.Storage {
+	storage := map[string]rest.Storage{}
+
+	// flow-schema
+	flowSchemaStorage, flowSchemaStatusStorage := flowschemastore.NewREST(restOptionsGetter)
+	storage["flowschemas"] = flowSchemaStorage
+	storage["flowschemas/status"] = flowSchemaStatusStorage
+
+	// priority-level-configuration
+	priorityLevelConfigurationStorage, priorityLevelConfigurationStatusStorage := prioritylevelconfigurationstore.NewREST(restOptionsGetter)
+	storage["prioritylevelconfigurations"] = priorityLevelConfigurationStorage
+	storage["prioritylevelconfigurations/status"] = priorityLevelConfigurationStatusStorage
+
+	return storage
+}
+
+func (p RESTStorageProvider) PostStartHook() (string, genericapiserver.PostStartHookFunc, error) {
+	systemPreset := SystemPresetData{
+		FlowSchemas:                 bootstrap.SystemFlowSchemas(),
+		PriorityLevelConfigurations: bootstrap.SystemPriorityLevelConfigurations(),
+	}
+	// TODO: default flow-schemas and priority levels
+	return PostStartHookName, systemPreset.EnsureSystemPresetConfiguration(), nil
+}
+
+type SystemPresetData struct {
+	FlowSchemas                 []*flowcontrolv1alpha1.FlowSchema
+	PriorityLevelConfigurations []*flowcontrolv1alpha1.PriorityLevelConfiguration
+}
+
+func (d SystemPresetData) EnsureSystemPresetConfiguration() genericapiserver.PostStartHookFunc {
+	return func(hookContext genericapiserver.PostStartHookContext) error {
+		flowcontrolClientSet := flowcontrolclient.NewForConfigOrDie(hookContext.LoopbackClientConfig)
```

Comment:
If I followed the logic correctly, this code gets run in extension servers --- where the loopback client config does NOT get to a main apiserver.

Reply:
No, registry/flowcontrol should not be running in an extension server.